

mm/vmscan.c: In function `kswapd':
mm/vmscan.c:1139: warning: no return statement in function returning non-void

Signed-off-by: Andrew Morton <akpm@osdl.org>
---

 25-akpm/mm/vmscan.c |    3 ++-
 1 files changed, 2 insertions(+), 1 deletion(-)

diff -puN mm/vmscan.c~kswapd-warning-fix mm/vmscan.c
--- 25/mm/vmscan.c~kswapd-warning-fix	2004-06-23 22:37:40.905413784 -0700
+++ 25-akpm/mm/vmscan.c	2004-06-23 22:40:46.518196352 -0700
@@ -1097,7 +1097,7 @@ out:
  * If there are applications that are active memory-allocators
  * (most normal use), this basically shouldn't matter.
  */
-int kswapd(void *p)
+static int kswapd(void *p)
 {
 	pg_data_t *pgdat = (pg_data_t*)p;
 	struct task_struct *tsk = current;
@@ -1136,6 +1136,7 @@ int kswapd(void *p)
 
 		balance_pgdat(pgdat, 0);
 	}
+	return 0;
 }
 
 /*
_
